src/userForm/UserForm.tsx

```diff
@@ -1,68 +1,143 @@
+    }
+  };
+
+  //tanstack/react-query hook to fetch the users
+  useQuery({
+    queryKey: ["searchUsernames", searchTerm?.trim()],
+    queryFn: getUsers,
+    enabled: false,
+  });
+
+  let timeoutId: NodeJS.Timeout;
+
+  //handle the input change, use debounce to wait for the user to stop typing, then only search
+  const handleInputChange = (inputValue: string) => {
+    clearTimeout(timeoutId);
+
+    timeoutId = setTimeout(() => {
+      setSearchTerm(inputValue);
+    }, 300);
+  };
+
+  const handleSelectChange = (selectedOption: SingleValue<Option>) => {
+    setSelectedOption(selectedOption as SingleValue<Option>);
+  };
 
   return (
-    <>
+    <div className="h-screen w-full flex flex-col justify-center items-center p-5 gap-5 lg:p-0">
+      <Drawer username={""} />
+
       <div
         id="pageContainer"
-        className="h-screen w-full flex flex-col justify-center items-center overflow-hidden overscroll-none absolute"
+        className="h-screen w-full flex flex-col justify-center items-center absolute lg:fixed lg:pl-[320px]"
       >
         <div
-          className="flex flex-col bg-off-white w-4/5 h-2/5 rounded-2xl shadow-3xl px-8 py-7 justify-around overflow-hidden lg:w-3/5"
+          className="flex flex-col bg-off-white w-4/5 h-2/5 rounded-2xl shadow-3xl px-8 py-7 justify-around lg:w-3/5"
           id="formContainer"
         >
           <div className="flex flex-col gap-1.5">
-            <h1 className="text-dark-text text-2xl lg:text-3xl">Find a User</h1>
-            <h2 className="text-lighter-text text-sm lg:text-base">
+            <h1 className="text-dark-text text-2xl lg:text-2xl">Find a User</h1>
+            <h2 className="text-ligher-text text-sm lg:text-base">
               Enter the Github username of the account you would like to view.
             </h2>
           </div>
           <Formik
             initialValues={{ username: "", rememberMe: false }}
-            validate={(values) => {
-              const errors: Partial<FormikValues> = {};
-              if (!values.username) {
-                errors.username = "Required";
-              } else if (!/^[A-Za-z0-9-]{3,15}$/i.test(values.username)) {
-                errors.username = "Invalid username address";
-              }
-              return errors;
-            }}
             onSubmit={(values, { setSubmitting }) => {
-              //when the form is submitted, the values are displayed in an alert
-              //write the values to json file if rememberMe is checked, and then that is the list of saved users that you will display on the saved menu
-              alert(JSON.stringify(values, null, 2));
+              console.log("Submitting form...");
+              console.log("Username:", selectedOption?.value);
+              console.log("Remember Me:", values?.rememberMe);
+              // Reset the form
+              navigate({ to: "/profile" });
+              //navigate({ to: '/posts/$postId', params: { postId } })
```

Comment:
Remember to remove commented out code (Or comment above if its for future use)

Reply:
It's for future use, I've added the comment now